### PR TITLE
ddl: fix a bug that upgrade the node when there is a paused job.

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/44225

Problem Summary:
Before the upgrade, it has a paused DDL job, it's upgrade will fail

### What is changed and how it works?
* Update judgment condition
* Move the callback from ddl/internal to ddl/util(Use it in test).

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: session/bootstraptest/bootstrap_upgrade_test.go [hunk 1]
```diff
@@ -293,6 +294,74 @@ func TestUpgradeVersionMockLatest(t *testing.T) {
...
+		require.NoError(t, err)
+		require.Len(t, rows, 1)
+		require.Equal(t, rows[0].GetString(2), "upgrade_tbl")
+
+		state := rows[0].GetString(11)
+		if state == "synced" {
+			suc = true
+			break
+		}
+		time.Sleep(time.Millisecond * 200)
```
Comment (Defined2014) [inline session/bootstraptest/bootstrap_upgrade_test.go:360]: Does it will cause UT too slow?

**Review by Defined2014 — APPROVED** [review]

File: session/bootstraptest/bootstrap_upgrade_test.go (hunk 1 above)
Comment (zimulala) [reply session/bootstraptest/bootstrap_upgrade_test.go:360]: In theory, no more than one retry is required to succeed. However, CI environment issues may require multiple retries just to stabilize the test, so multiple retries are added

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**